### PR TITLE
I#8 implement reselect

### PR DESCRIPTION
**Pull request by PabloAlice**

### CODE REVIEW

**Comment by PabloAlice** [comment]
Well, I didn't want to get out of scope making changes. Actually, I think these warnings already was there, you don't see them in the Playground deployment because it was built with production as Node env, that means React removed prop types checking for performance reasons.

Anyway, I have to increase the test coverage, so I will fix these warnings too.

**Comment by JirkaChadima** [comment]
They don't show locally on develop either... So they were definitely not
there :)

**Comment by skkr** [comment]
Guys, I'm guessing that the issue is caused because there are anchor tags in the markdown used in the cards, is that so? The cards are wrapped in anchor tags, so if there are links in the card body, it will show a warning. 

If that's the case I'd recommend to remove the links from the markdown. Because there is a design decision of not put links in them, except by the ones on the footer (which aren't really links). If that's not possible, the anchor on the .card could be replicated with JS but we'll loose the hover animation on the cards. If this is the issue and there is no other option, let me know.

**Comment by PabloAlice** [comment]
> Guys, I'm guessing that the issue is caused because there are anchor tags in the markdown used in the cards, is that so? The cards are wrapped in anchor tags, so if there are links in the card body, it will show a warning.
> 
> If that's the case I'd recommend to remove the links from the markdown. Because there is a design decision of not put links in them, except by the ones on the footer (which aren't really links). If that's not possible, the anchor on the .card could be replicated with JS but we'll loose the hover animation on the cards. If this is the issue and there is no other option, let me know.

**Comment by JirkaChadima** [comment]
@skkr @PabloAlice We can't get rid of anchors in cards. They might be part of the hotel description which we have no control of.

File: src/selectors/hotels.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export function getNextHotel(state) {
+  return state.hotels.next;
+}
+
+export function areHotelsInitialized(state) {
+  return state.hotels.hotelsInitialized;
+}
+
+export function isLoadingMore(state) {
+  return state.hotels.isLoadingMore;
```
Comment (JirkaChadima) [inline src/selectors/hotels.js]: @PabloAlice This is where the bug with isLoadingMore was introduced. The originally passed property was reading `state.hotels.hotelsLoading` (https://github.com/windingtree/wt-hotel-explorer/pull/48/files#diff-79e69c769127d3be6b5dd17dffdd5f83L91)

File: test/selectors/hotels.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  };
+  const identity = state => state.hotels.list;
+  let state;
+  beforeEach(() => {
+    state = {
+      hotels: {
+        list: [{ id: 1, name: 'someHotel' }],
+        next: 'urlToNext',
+        hotelsInitialized: false,
+        isLoadingMore: true,
```
Comment (JirkaChadima) [inline test/selectors/hotels.spec.jsx]: This is actually incompatible with the `defaultState` in `reducers/hotels.jsx` So the tests might pass, but the actual code does not work.

File: src/selectors/hotels.js (hunk 1 above)
Comment (PabloAlice) [reply src/selectors/hotels.js]: Awesome! I was wrong about this warning, thank you for the review!

**Comment by PabloAlice** [comment]
@JirkaChadima I see, can we open a new issue to discuss it?

**Comment by JirkaChadima** [comment]
> @JirkaChadima I see, can we open a new issue to discuss it?

Yeah, sure. I initially said that the anchor problem probably is unrelated :) It's caused by CRON hotel on `0x037Ee5a21F662720bDD535620442C0A8B3E21FF7`

**Pull request merged**